Warn about practices that lead to flaky tests (#22389)
prestodb/presto: CONTRIBUTING.md

* Categorize errors when throwing an exception
* **Tests**
    * Avoid adding `Thread.sleep` in tests--these can fail due to environmental conditions, such as garbage collection or noisy neighbors in the CI environment.
    * Do not use random values in tests. All tests should be reproducible.

>     * Be extremely careful when storing test data in fields. This is really

Reviewer: ```suggestion
    * Be careful of storing test data in fields, because doing so is
```

Author: done

      only appropriate for constant, immutable values. Most Presto tests
      run with TestNG which, unlike JUnit, does not create a new object for
      each test. Shared instance fields can make tests tightly coupled,

>       order dependent, and flaky. If you do use instance fields, 

Reviewer: ```suggestion
      order dependent, and flaky. If you use instance fields, 
```
